manifests/1.3.5/opensearch-1.3.5.yml

```diff
@@ -44,3 +44,14 @@ components:
     checks:
       - gradle:properties:version
       - gradle:dependencies:opensearch.version
+    platforms:
+      - linux
```

Comment:
Should the platform also include darwin like 1.3.4 
https://github.com/opensearch-project/opensearch-build/blob/381aebc033b2a62b439b977c7bbed8e5228d91cb/manifests/1.3.4/opensearch-1.3.4.yml#L44-L46

Reply:
@zelinh since now we actually have darwin agents added to the jenkins, k-nn had issues to build with darwin
https://github.com/opensearch-project/k-NN/issues/493, hence I have removed it from `platforms`.

Reply:
Build error: https://build.ci.opensearch.org/job/distribution-build-opensearch/5927/execution/node/183/log/